pkg/resource/model_package/testdata/test_suite.yaml

```diff
@@ -1,17 +1,251 @@
+            - operation: CreateModelPackageWithContext
+              output_fixture: "sdkapi/create/create_success.json"
+        invoke: Create
+        expect:
+          latest_state: "v1alpha1/create/observed/success_after_create_variation.yaml"
+          error: nil
+      - name: "Create=ValidVersioned"
+        description: "Create a new versioned Model package successfully (ARN in status)."
+        given:
+          desired_state: "v1alpha1/create/desired/success_before_create_versioned.yaml"
+          svc_api:
+            - operation: CreateModelPackageWithContext
+              output_fixture: "sdkapi/create/create_success_versioned.json"
+        invoke: Create
+        expect:
+          latest_state: "v1alpha1/create/observed/success_after_create_versioned.yaml"
+          error: nil
+  - name: "Model package readOne tests"
+    description: "Testing the readOne operation"
+    scenarios:
+      - name: "ReadOne=MissingRequiredField"
+        description: "Testing readOne when required field is missing. No API call is made and returns error."
+        given: 
+          desired_state: "v1alpha1/readone/desired/missing_required_field.yaml"
+        invoke: ReadOne
+        expect:
+          error: "resource not found"
+      - name: "ReadOne=NotFound"
+        description: "Testing readOne when Describe fails to find the resource on SageMaker"
+        given: 
+          desired_state: "v1alpha1/readone/desired/after_create.yaml"
+          svc_api:
+            - operation: DescribeModelPackageWithContext
+              error:
+                code: ValidationException
+                message: "model package xgboost-unversiosned-model-package does not exist."
+        invoke: ReadOne
+        expect:
+          error: "resource not found"
+      - name: "ReadOne=Fail"
+        description: "This test checks if the condition is updated if describe fails and readOne returns error"
+        given: 
+          desired_state: "v1alpha1/readone/desired/after_create.yaml"
+          svc_api:
+            - operation: DescribeModelPackageWithContext
+              error:
+                code: ServiceUnavailable
+                message: "Server is down"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/error_on_describe.yaml"
+          error: "ServiceUnavailable: Server is down\n\tstatus code: 0, request id: "
+      - name: "ReadOne=Pending"
+        description: "Testing readone right after create, the status should be in Pending with Resource synced being false."
+        given: 
+          desired_state: "v1alpha1/readone/desired/pending_after_create.yaml"
+          svc_api:
+            - operation: DescribeModelPackageWithContext
+              output_fixture: "sdkapi/describe/describe_pending_after_create.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/pending_after_create.yaml"
+      - name: "ReadOne=Deleting"
```

Comment:
Nice, I don't suppose other resources have this one

Reply:
Haven't seen other resources with pending state :)